src/components/Task.js

```diff
@@ -30,39 +32,41 @@ const TaskCheckbox = styled.div`
 `
 
 
-class Task extends React.Component {
+const Task = (props) => {
+    const { dispatch } = useContext(TaskContext);
+    const {id, text, status} = props.task;
 
-
-    ClickHandler = (status, id) => {
+    const ClickHandler = (status, id) => {
         if (status === "cancel") {
-            this.props.setTaskStatus("", id);
+            dispatch(updateStatus("", id));
         } else if (status === "complete") {
-            this.props.setTaskStatus("cancel", id);
+            dispatch(updateStatus("cancel", id));
```

Comment:
I suppose here you have mistaken. In conditional expression `status === "complete"` but in a action you pass `cancel`